src/RestApi/StoreApi/Utilities/CartController.php

```diff
@@ -209,6 +209,24 @@ public function get_cart_coupons( $callback = null ) {
 		return $callback ? array_filter( wc()->cart->get_applied_coupons(), $callback ) : array_filter( wc()->cart->get_applied_coupons() );
 	}
 
+	/**
+	 * Selects a shipping rate.
+	 *
+	 * @param array $rate_ids Shipping rate ids.
+	 */
+	public function select_shipping_rate( $rate_ids ) {
+		$cart                    = $this->get_cart_instance();
+		$chosen_shipping_methods = WC()->session->get( 'chosen_shipping_methods' ) ? WC()->session->get( 'chosen_shipping_methods' ) : array();
+		$posted_shipping_methods = $rate_ids ? wc_clean( wp_unslash( $rate_ids ) ) : array();
```

Comment:
I think I may have mentioned this earlier; you're trusting user data here - we need to apply some form of sanitization to $rate_ids, probably in `select_shipping` method rather than this controller.

Reply:
would the sanitization of `wc_clean` and `wp_unslash` be enough?

Reply:
This is an array of data, so test the shape I think? And yes those.

Reply:
it is an unindexed array of strings, not sure how I can test the shape of that (checking if all elements are strings could be one) but I think wc_clean does that.